src/inverse_cai/data/annotated_pairs_format.py

```diff
@@ -20,20 +20,35 @@
 DEFAULT_ANNOTATOR_DESCRIPTION = (
     "Default annotator from original dataset (from column `preferred_text`)"
 )
-FORMAT_VERSION = "1.0"
+FORMAT_VERSION = "2.0"
 DEFAULT_ANNOTATOR_TYPE = "unknown"
 DEFAULT_PREFERENCE_KEY = "pref"
 DEFAULT_PREFERENCE_COLUMN = "preferred_text"
 
+PREFERENCE_ALIAS_MAPPING = {
+    "a": "a",
+    "b": "b",
+    "text_a": "a",
+    "text_b": "a",
+}
+
 
 def hash_string(s: str) -> str:
     """Create a shortened hash of a string."""
     return hashlib.md5(s.encode("utf-8")).hexdigest()[:8]
 
 
-def hash_comparison(text_a: str, text_b: str, prompt: Optional[str]) -> str:
-    """Create a hash ID for a comparison based on its content."""
-    combined = f"{text_a}|{text_b}|"
+def hash_comparison(
+    response_a: Dict[str, Any], response_b: Dict[str, Any], prompt: Optional[str]
+) -> str:
+    """Create a hash ID for a comparison based on its content.
+
+    Args:
+        response_a: A dictionary with at least a 'text' key
+        response_b: A dictionary with at least a 'text' key
+        prompt: An optional prompt string
+    """
+    combined = f"{response_a['text']}|{response_b['text']}|"
```

Comment:
There is a special case we want to account for: when three models produce the same response to a prompt, then this leads to identical hashes for 3 three different model comparisons - even though they contain different information for the sake of comparing models. This actually happens more frequently than one might expect (e.g. a prompt that just asks for an "ok"). I think we should add the model names into the hash to account for this special case and easily keep track of these different scenarios. If think the efficiency loss is relatively low, as these scenarios are not exactly frequent either.

Reply:
What do you think?